src/transformers/modeling_dpr.py

```diff
@@ -0,0 +1,541 @@
+            They correspond to the combined `input_ids` from `(question + context title + context content`).
+        **start_logits**: ``torch.FloatTensor`` of shape ``(n_passages, sequence_length)``
+            Logits of the start index of the span for each passage.
+        **end_logits**: ``torch.FloatTensor`` of shape ``(n_passages, sequence_length)``
+            Logits of the end index of the span for each passage.
+        **relevance_logits**: `torch.FloatTensor`` of shape ``(n_passages, )``
+            Outputs of the QA classifier of the DPRReader that corresponds to the scores of each passage
+            to answer the question, compared to all the other passages.
+
+
+    Examples::
+
+        tokenizer = DPRReader.from_pretrained('dpr-base-uncased')
+        model = DPRModel.from_pretrained('dpr-reader-base')
+        question_and_titles_ids = [
+            torch.tensor(tokenizer("Hello, is my dog cute ?", "Dog cuteness"))
+            ]  # One tensor per passage. It corresponds to the concatenation of the question and the context title.
+        texts_ids = [
+            torch.tensor(tokenizer("Hello, my dog is definitely cute", add_special_tokens=False))
+            ]  # One tensor per passage. It corresponds to the context text in which we're looking for the answer.
+        outputs = model(question_and_titles_ids, texts_ids)
+        start_logits = outputs[1]  # The logits of the start of the span
+        end_logits = outputs[2]  # The logits of the end of the span
+
+    """
+
+    def __init__(self, config: DPRConfig):
+        super().__init__(config)
+        self.config = config
+        self.span_predictor = DPRSpanPredictor(config)
+        self.init_weights()
+
+    def forward(self, question_and_titles_ids: List[Tensor], texts_ids: List[Tensor],) -> Tuple[Tensor, ...]:
+        assert len(question_and_titles_ids) == len(
+            texts_ids
+        ), "There should be as many `question_and_titles_ids` than `texts_ids` but got sizes {}!={}".format(
+            len(question_and_titles_ids), len(texts_ids)
+        )
+        device = question_and_titles_ids[0].device
+        n_contexts = len(question_and_titles_ids)
+        input_ids = torch.ones((n_contexts, self.config.sequence_length), dtype=torch.int64, device=device) * int(
+            self.config.pad_token_id
+        )
+        for i in range(n_contexts):
+            question_and_title_ids = question_and_titles_ids[i]
+            text_ids = texts_ids[i]
+            len_question_and_title = question_and_title_ids.shape[1]
+            _, len_txt = text_ids.size()
+            input_ids[i, 0:len_question_and_title] = question_and_title_ids
+            input_ids[i, len_question_and_title : len_question_and_title + len_txt] = text_ids
+        input_ids = input_ids.unsqueeze(0)
+        attention_mask = input_ids != self.config.pad_token_id
+        attention_mask = attention_mask.to(device=device)
```

Comment:
All this should be handled by the tokenizer, no?

Reply:
I'll keep it for the next PR (cf other comment)